controllers/UsersController.js

```diff
@@ -1,92 +1,92 @@
+            try {
+                await users_model.update(
+                    'status=\'verified\'',
+                    `id=${id}`
+                );
+                UsersController.get_user(req, res);
+            }
+            catch (e) { return InternalServerError(req, res, e); }
+        })();
     },
-    
+
     get_users: (req, res) => {
         const { status } = req.query;
-        
-        if (status) {
-            const data = users.filter(user => (user.status === status));
-            res.status(200).json({ data });
-        }
-        else {
-            const data = [];
-            users.map(user => data.push(user));
-            res.status(200).json({ data });
-        }
+        (async () => {
+            try {
+                let data;
+                if (status) {
+                    data = await users_model.select(
+                        `id, email, password, firstname,
+                    lastname, phone, status, address`,
+                        `status='${status}'`
+                    );
+                }
+                else {
+                    data = await users_model.select(
+                        `id, email, password, firstname,
+                        lastname, phone, status, address`,
+                    );
+                }
+                return res.status(200).json({ data: data.rows });
+            }
+            catch (e) { return InternalServerError(req, res, e); }
+        })();
     },
 
-    get_user: (req, res) => {
+    update_user: (req, res) => {
         const { id } = req.params;
-        const user = users.find(user => (user.id === id));
-        if (user === undefined) {
-            res.status(404).json({ error: `User with id ${id} not found` });
-        }
-        else res.status(200).json({ data: user });
-    },
+        const { firstname, lastname, phone, home, office } = req.body;
 
-    update_user: [
-        body('firstName')
-            .not().isEmpty().withMessage('First name cannot be empty'),
-        body('lastName')
-            .not().isEmpty().withMessage('Last namecannot be empty'),
-        body('phone')
-            .not().isEmpty().withMessage('Phone number cannot be empty')
-            .matches(/^0\d{10}$/).withMessage(
-                'Wrong number format: E.G. 07012345678'),
-        body('home')
-            .not().isEmpty().withMessage('Home address cannot be empty'),
-        body('office')
-            .not().isEmpty().withMessage('Office address cannot be empty'),
-
-        sanitizeBody('firstName').trim().escape(),
-        sanitizeBody('lastName').trim().escape(),
-        sanitizeBody('phone').trim().escape(),
-        sanitizeBody('home').trim().escape(),
-        sanitizeBody('office').trim().escape(),
-
-        (req, res) => {
-            const errors = validationResult(req);
-            if (!errors.isEmpty()) {
-                return res.status(422).json({ errors: errors.array() });
+        (async () => {
```

Comment:
Make the outer function async you won't have a need for this IIFE

Reply:
Done